### PR TITLE
Fixed courseListingEntries/index.ts to parse class schedule information

### PR DESCRIPTION
**Pull request by keithradford**
Modified courseListingEntries/index.ts so the info it scrapes from BAN1P into the object it returns matches how the test in index.test.ts was written up.
I modified types.ts to be a bit more general and we can go back and fix that if possible, but it wasn't liking when it was supposed to be for example `type sectionType` which is `'lecture' | 'lab' | 'tutorial'` since it doesn't know exactly what it's parsing from the HTML doc. I think it would be good if we can use these types to ensure data is in the right spot.
Also I would like to go back and make this code a bit more modular, it's very regexy and would be nice if we could cut down on the RegEx and use some more general statements. I have some ideas in mind but this works for the time being.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "dotenv": "^8.2.0",
     "got": "^11.7.0",
     "jest": "^26.5.3",
+    "moment": "^2.29.1",
```
Comment (aomi) [inline package.json:20]: Can we use a newer library for this? MomentJS works fine but I believe there is more modern alternatives and we should try and use newer ones.

File: src/pages/courseListingEntries/__tests__/index.test.ts [hunk 2]
```diff
@@ -15,14 +15,33 @@ describe('Class Schedule Listing Parser', () => {
...
+    expect(parsed[0].crn).toBe('10435');
+    expect(parsed[0].associatedTerm).toStrictEqual({ start: '201909', end: '201912' });
+    expect(parsed[0].registrationDates).toStrictEqual({ start: 'Jun 17, 2019', end: 'Sep 20, 2019' });
+    expect(parsed[0].campus).toBe('in-person');
+    expect(parsed[0].instructionalMethod).toBe('in-person');
+    expect(parsed[0].sectionType).toBe('lecture');
+    expect(parsed[0].credits).toBe('1.500');
+    expect(parsed[0].levels).toStrictEqual(['law', 'undergraduate']);
+    expect(parsed[0].schedule[0].type).toBe('Every Week');
+    expect(parsed[0].schedule[0].instructors).toBe('Neil   Burford (P), Cornelia   Bohne , Genevieve Nicole  Boice , J. Scott   McIndoe');
```
Comment (aomi) [inline src/pages/courseListingEntries/__tests__/index.test.ts:45]: Can we make the `instructors` property a `string[]`?

File: src/pages/courseListingEntries/index.ts [hunk 3]
```diff
@@ -12,12 +13,12 @@ export const classScheduleListingExtractor = async ($: cheerio.Root): Promise<Se
   try {
     const sections: Section[] = [];
     const sectionEntries = $(`table[summary="This layout table is used to present the sections found"]>tbody>tr`);
-    for (let sectionIdx = 0; sectionIdx < sectionEntries.length; sectionIdx += 2) {
+    for (let sectionIdx = 0; sectionIdx < sectionEntries.length; sectionIdx += 2) { //why += 2?
```
Comment (aomi) [inline src/pages/courseListingEntries/index.ts:16]: I believe that's for the data is formatted with tables but if there's a better way to do this we can address it in a future PR work.

**Comment by aomi** [comment]
@keithradford Can the test be fixed or is that unrelated to this fix?

**Comment by keithradford** [comment]
> 
> 
> @keithradford Can the test be fixed or is that unrelated to this fix?

@aomi 
Unrelated. I fixed the course listing parser and the error is coming from the detailed course parser. I'll tackle that next

File: src/pages/courseListingEntries/index.ts [hunk 4]
```diff
@@ -33,29 +34,94 @@ export const classScheduleListingExtractor = async ($: cheerio.Root): Promise<Se
...
+      // Check if online campus or in-person campus
+      // Might change this because in the HTML it's either: online or main campus (might be other campuses too)
+      if(/online/i.test(sectionInfo[4]) === true){
+        section.campus = 'online';
+      }
+      else{
+        section.campus = 'in-person';
+      }
+
+      section.sectionType = /(lecture|lab|tutorial)/i.exec(sectionInfo[5])![1].toLowerCase();
```
Comment (aomi) [inline src/pages/courseListingEntries/index.ts:112]: you can add `as SectionType` for example to cast it so the types don't have issues.

**Review by aomi — APPROVED** [review]

**Pull request merged**